### PR TITLE
New events

### PR DESCRIPTION
**Pull request by NAlexandrov**
https://hemerajs.github.io/hemera/docs/error-event.html

### CODE REVIEW

**Comment by StarpTech** [comment]
Good catch! Thank you.

File: packages/hemera/index.d.ts [hunk 1]
```diff
@@ -444,6 +444,18 @@ declare class Hemera<Request, Response> {
       instance: Hemera<Hemera.ServerRequest, Hemera.ServerResponse>
     ) => void
   ): Hemera<Request, Response>
+  on(
```
Comment (StarpTech) [inline packages/hemera/index.d.ts]: The `this` context is of type `Hemera`

Comment (StarpTech) [reply packages/hemera/index.d.ts]: Could you add it to all event handlers? https://www.typescriptlang.org/docs/handbook/functions.html#this-parameters

Comment (StarpTech) [reply packages/hemera/index.d.ts]: Something like:

```js
  on(
    event: 'clientResponseError',
    handler: (
      this: Hemera<Hemera.ClientRequest, Hemera.ClientResponse>,
      error: Error
    ) => void
  ): void
```

Comment (StarpTech) [reply packages/hemera/index.d.ts]: I will do it in a separate PR.

Comment (NAlexandrov) [reply packages/hemera/index.d.ts]: done #216

Comment (StarpTech) [reply packages/hemera/index.d.ts]: Hi @NAlexandrov my changes are already landed.

**Pull request merged**